# 중간고사 대체 실습과제 — 문제 1

## 기획의 이해와 6단계 프로세스

| 항목 | 내용 |
|------|------|
| **관련 챕터** | Ch01. 기획이란 무엇인가? |
| **핵심 개념** | 기획의 정의, 분석/계획/전략 구분, 6단계 기획 프로세스 |
| **배점** | 20점 |
| **제출** | 이 노트북(.ipynb)을 실행 결과 포함하여 제출 |

---

## 시나리오

당신은 건강 도시락 구독 서비스 **"프레시밀"**의 기획팀 신입사원입니다.
최근 3개월간 구독 해지율이 15%에서 28%로 급증하고 있습니다.

| 항목 | 수치 |
|------|------|
| 현재 구독자 수 | 3,200명 |
| 월 구독료 | 5만 원 |
| 최근 해지율 | 28% (기존 15%) |
| 월 신규가입 | 120명 |
| 고객 불만 Top 3 | ①배송 지연(35%) ②메뉴 단조(28%) ③가격(22%) |

In [1]:
# 환경설정
import numpy as np
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [2]:
# 데이터 로드
monthly = pd.DataFrame({
    '월': ['10월', '11월', '12월', '1월', '2월', '3월'],
    '구독자수': [3800, 3650, 3500, 3400, 3300, 3200],
    '해지율_pct': [15, 18, 22, 25, 27, 28],
    '배송지연건수': [45, 62, 88, 110, 125, 140],
    '고객만족도': [4.2, 4.0, 3.7, 3.4, 3.2, 3.0]
})

complaint = pd.DataFrame({
    '불만유형': ['배송 지연', '메뉴 단조', '가격 불만', '포장 품질', '기타'],
    '비율_pct': [35, 28, 22, 10, 5]
})

print(monthly.to_string(index=False))
print()
print(complaint.to_string(index=False))

  월  구독자수  해지율_pct  배송지연건수  고객만족도
10월  3800       15      45    4.2
11월  3650       18      62    4.0
12월  3500       22      88    3.7
 1월  3400       25     110    3.4
 2월  3300       27     125    3.2
 3월  3200       28     140    3.0

 불만유형  비율_pct
배송 지연      35
메뉴 단조      28
가격 불만      22
포장 품질      10
   기타       5


---

## 과제 1. 개념 이해 (6점)

프레시밀 사례를 활용하여 답하세요.

### 요구사항

1. **(3점)** "분석", "기획", "전략"의 차이를 프레시밀 사례에 적용하여 **각각 1문장**으로 설명하세요.

2. **(3점)** 프레시밀의 해지율 대응은 "전략기획", "정책기획", "사업기획" 중 어디에 해당하는지 고르고 이유를 서술하세요.

### ✏️ 과제 1 답안

**1. 분석/기획/전략 구분**

- 분석: 프레시밀이 해지 원인을 데이터로 확인하는 단계를 말한다.
- 기획: 파악된 문제를 바탕으로 실제로 실행할 해결 방법을 구체화하는 과정이다.
- 전략: 프레시밀이 여러 해결 방법 중에서 가장 효과적인 방향을 선택한다.

**2. 기획 유형**

- 유형: 사업기획
- 이유: 고객 이탈을 줄이기 위해 실제로 실행할 방법을 적용하는 과정이기 때문에, 실무 중심의 사업기획이라고 볼 수 있다.

---

## 과제 2. 6단계 기획 프로세스 적용 (8점)

```
1. 문제 정의  →  2. 원인 분석  →  3. 대안 개발
→  4. 의사결정  →  5. 실행  →  6. 학습
```

### 요구사항

1. **(3점)** 제공된 데이터를 시각화하세요:
   - (a) 월별 해지율 추이 라인 차트
   - (b) 불만 유형 비율 파이 차트

2. **(5점)** 6단계 프로세스 중 **1~4단계**를 아래 양식에 작성하세요:
   - 1단계(문제 정의): 핵심 문제를 1~2문장으로
   - 2단계(원인 분석): 데이터에서 확인되는 원인 2개
   - 3단계(대안 개발): 해결 대안 3개
   - 4단계(의사결정): 가장 효과적인 대안 1개 선택 + 이유

In [3]:
# ✏️ 과제 2-1: 시각화
# (a) 월별 해지율 라인 차트
import plotly.graph_objects as go

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=monthly['월'],
    y=monthly['해지율_pct'],
    mode='lines+markers',
    line=dict(color='black'),
    marker=dict(
        color='white',
        line=dict(color='black', width=2)
    )
))

fig.update_layout(
    title='월별 해지율 추이',
    font=dict(color='black')
)

fig.show()
# (b) 불만 유형 파이 차트
import plotly.express as px

fig = px.pie(
    complaint,
    names='불만유형',
    values='비율_pct',
)

fig.update_traces(
    textfont=dict(color='black'),
    marker=dict(line=dict(color='black', width=1))
)

fig.update_layout(
    font=dict(color='black')
)

fig.show()

### ✏️ 과제 2 해석 (이 셀에 직접 타이핑)

시각화 결과를 보고 아래에 답하세요.

- 해지율 추이에서 가장 눈에 띄는 변화 시점은? 그 이유는 무엇이라 추측하는가?
- 불만 유형 중 가장 비율이 높은 항목이 해지율과 어떻게 연결되는가? (1~2문장)

### ✏️ 과제 2-2: 6단계 프로세스 (1~4단계)

**1단계. 문제 정의:** 월별 데이터를 보면 해지율이 10월 약 15%에서 3월 28%까지 빠르게 증가한 것을 확인할 수 있다.
이로 인해 고객 이탈이 심해지고 있으며, 서비스에 대한 신뢰도가 낮아지고 있는 것이 핵심 문제라고 생각한다.

**2단계. 원인 분석:**
- 원인 1: 배송 지연이 전체 불만의 35%를 차지하고 있어 고객 이탈의 주요 원인으로 보인다.
- 원인 2: 메뉴가 단조롭고 가격에 대한 불만이 함께 나타나면서 서비스 대비 가치가 낮다고 느끼는 점도 영향을 준 것으로 보인다.

**3단계. 대안 개발:**

| 대안 | 내용 |
|------|------|
| 대안 1 |배송 지연 문제를 줄이기 위해 물류 시스템을 개선하고 배송 안정성을 높일 필요가 있다. |
| 대안 2 |메뉴를 다양화하여 고객이 느끼는 서비스 만족도를 높일 수 있다. |
| 대안 3 |할인이나 멤버십 혜택을 통해 가격 부담을 줄이고 고객 유지 효과를 높일 수 있다. |

**4단계. 의사결정:** 제일 효과적인 대안은 배송 시스템 개선이라고 생각한다.
그 이유는 배송 지연이 가장 큰 불만 원인이기 때문에 해지율 감소에 가장 직접적인 영향을 줄 수 있다고 판단했기 때문이다.

---

## 과제 3. 간단한 시뮬레이션 (6점)

현재 상태(구독자 3,200명, 해지율 28%, 신규 120명/월)로 **6개월 후 구독자 수**를 계산하세요.

### 요구사항

1. **(3점)** 아래 두 시나리오의 6개월간 구독자 수 변화를 계산하세요:
   - 시나리오 A: 현상 유지 (해지율 28%)
   - 시나리오 B: 배송 개선으로 해지율 20%로 감소

2. **(3점)** 두 시나리오의 구독자 수 추이를 **라인 차트**로 비교하고, 6개월 후 차이를 `print()`로 출력하세요.

In [4]:
# ✏️ 과제 3 풀이
# 시나리오 A vs B 시뮬레이션 + 라인 차트 + 결론 print()

# 초기값
initial_subscribers = 3200
new_users = 120
months = 6

# 시나리오 A (해지율 28%)
churn_A = 0.28

# 시나리오 B (해지율 20%)
churn_B = 0.20

subs_A = [initial_subscribers]
subs_B = [initial_subscribers]

# 계산
for i in range(months):
    next_A = subs_A[-1] * (1 - churn_A) + new_users
    next_B = subs_B[-1] * (1 - churn_B) + new_users
    
    subs_A.append(next_A)
    subs_B.append(next_B)

# x축 (개월)
x = list(range(0, months+1))

# 그래프
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=x,
    y=subs_A,
    mode='lines+markers',
    name='시나리오 A',
    line=dict(color='black'),
    marker=dict(color='white', line=dict(color='black', width=2))
))

fig.add_trace(go.Scatter(
    x=x,
    y=subs_B,
    mode='lines+markers',
    name='시나리오 B',
    line=dict(color='gray'),
    marker=dict(color='white', line=dict(color='black', width=2))
))

fig.update_layout(
    title='구독자 수 변화 (6개월)',
    xaxis_title='개월',
    yaxis_title='구독자 수 (명)',
    font=dict(color='black')
)

fig.show()

# 6개월 후 차이 출력
difference = subs_B[-1] - subs_A[-1]
print(f"시나리오 A (해지율 28%) - 6개월 후 구독자 수: {int(subs_A[-1])}명")
print(f"시나리오 B (해지율 20%) - 6개월 후 구독자 수: {int(subs_B[-1])}명")
print(f"차이: {int(difference)}명")

시나리오 A (해지율 28%) - 6개월 후 구독자 수: 814명
시나리오 B (해지율 20%) - 6개월 후 구독자 수: 1281명
차이: 466명


### ✏️ 과제 3 해석 (이 셀에 직접 타이핑)

- 시나리오 A와 B의 6개월 후 구독자 수 차이는 몇 명인가?
- 이 차이가 **매출(월 구독료 5만 원)**로 환산하면 얼마인가?
- 이 결과가 경영진에게 시사하는 바를 1문장으로 작성하세요.

그래프를 보면 해지율이 높은 시나리오 A는 시간이 지날수록 구독자 수가 빠르게 줄어드는 것을 확인할 수 있다. 반면 시나리오 B는 감소 폭이 비교적 작아서 더 많은 구독자를 유지하는 모습을 보인다. 6개월 후 기준으로 약 466명 정도 차이가 발생하고, 매출로 보면 약 2,330만 원 정도의 차이가 나는 것으로 나타난다. 따라서 해지율을 낮추는 것이 고객 유지뿐만 아니라 매출에도 중요한 영향을 준다고 생각한다.

---

## 채점 기준

| 과제 | 배점 | 채점 포인트 |
|------|------|------------|
| 과제 1. 개념 이해 | 6점 | 분석/기획/전략 구분(3) + 기획유형(3) |
| 과제 2. 6단계 프로세스 | 8점 | 시각화(3) + 프로세스 적용(5) |
| 과제 3. 시뮬레이션 | 6점 | 계산(3) + 시각화+결론(3) |
| **합계** | **20점** | |

---

## 참고: 예상 정답값

**과제 1 예시:**
- 분석: "과거 6개월 해지 데이터를 분석하여 배송 지연이 주 원인임을 파악하는 것"
- 기획: "해지율을 15%로 되돌리기 위한 구체적 행동 계획을 설계하는 것"
- 전략: "프리미엄 건강식 시장의 1위라는 방향성을 설정하는 것"

**과제 3 시뮬레이션 (6개월 후 근사치):**

| 시나리오 | 6개월 후 구독자 수 |
|---------|-------------------|
| A. 현상 유지 (28%) | ~816명 |
| B. 배송 개선 (20%) | ~1,283명 |

- 배송 개선 시 약 467명 더 유지 가능